src/Sarif/Visitors/ValidatingVisitor.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis.Sarif.Writers;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class ValidatingVisitor : SarifRewritingVisitor, IDisposable
+    {
+        private static Rule InvalidUri = new Rule()
+        {
+            Id = "SARIF001",
+            ShortDescription = "Uri was not valid.",
+            Name = nameof(InvalidUri)
+        };
+
+        private SarifLogger _sarifLog;
+
+        public ValidatingVisitor(TextWriter writer)
+        {
+            var tool = Tool.CreateFromAssemblyData();
+            tool.Name = "SarifValidatingVisitor";
+
+            _sarifLog = new SarifLogger(writer, verbose: true, tool: tool);
+        }
+
+        public override FileData VisitFileData(FileData node)
+        {
+            ValidateUri(node.Uri);
+            return base.VisitFileData(node);
+        }
+
+        public override PhysicalLocation VisitPhysicalLocation(PhysicalLocation node)
+        {
+            ValidateUri(node.Uri);
+            return base.VisitPhysicalLocation(node);
+        }
+
+        public override StackFrame VisitStackFrame(StackFrame node)
+        {
+            ValidateUri(node.Uri);
+            return base.VisitStackFrame(node);
+        }
+
+        public override Rule VisitRule(Rule node)
+        {
+            ValidateUri(node.HelpUri);
+            return base.VisitRule(node);
+        }
+
+        public override FileChange VisitFileChange(FileChange node)
+        {
+            ValidateUri(node.Uri);
+            return base.VisitFileChange(node);
+        }
+
+        private void ValidateUri(Uri uri)
+        {
+            if (!uri.IsWellFormedOriginalString())
+            {
```

Comment:
I don't get this validation. It will, for example, trip on new Uri(@"C:\MyPath\MyFile.cs"). I would think that producers would do that for convenience. This will force them to do new Uri(new Uri(@"c:\MyPath\MyFile.cs").AbsoluteUri) or am I missing something?